Markdown/Default.sublime-keymap

```diff
@@ -0,0 +1,41 @@
+[
+    // Auto-pair asterisks
+    { "keys": ["*"], "command": "insert_snippet", "args": {"contents": "*$0*"}, "context":
+        [
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selector", "operator": "equal", "operand": "text.html.markdown" },
+            { "key": "selection_empty", "operator": "equal", "operand": true, "match_all": true },
+            { "key": "following_text", "operator": "regex_contains", "operand": "^(?:\t| |\\)|]|\\}|>|$)", "match_all": true },
+        ]
+    },
+    {
+        "keys": ["*"],
+        "command": "insert_snippet",
+        "args": {"contents": "*${0:$SELECTION}*"},
+        "context": [
+            { "key": "selector", "operator": "equal", "operand": "text.html.markdown" },
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selection_empty", "operator": "equal", "operand": false, "match_all": true },
+        ]
+    },
+
+    // Auto-pair backticks
+    { "keys": ["`"], "command": "insert_snippet", "args": {"contents": "`$0`"}, "context":
+        [
+            { "key": "setting.auto_match_enabled", "operator": "equal", "operand": true },
+            { "key": "selector", "operator": "equal", "operand": "text.html.markdown" },
+            { "key": "selection_empty", "operator": "equal", "operand": true, "match_all": true },
+            { "key": "following_text", "operator": "regex_contains", "operand": "^(?:\t| |\\)|]|\\}|>|$)", "match_all": true },
+        ]
+    },
```

Comment:
In the case of `"`, you can press <kbd>backspace</kbd> and the extra inserted `"` (totalling `"|"`, where `|` is the cursor) also is removed. This behavior seems to be missing for \`.

Reply:
I'd rather expect to toggle `bold` and `italic` style via the same key binding.

That said, Markdown may need several such input helpers which I'd like to know whether they are wanted to be part of this core package first.

Otherwise an up-to-date "SmartMarkdown" package may do the job as well. MarkdownEditing has much to offer, too, but may need to be stipped down a bit to remove no longer needed stuff.

Reply:
> In the case of `"`, you can press backspace and the extra inserted `"` (totalling `"|"`, where `|` is the cursor) is also removed. This behavior seems to be missing for `.

Fixed.

Reply:
> I'd rather expect to toggle `bold` and `italic` style via the same key binding.

I can see the argument for `cmd+I` & `cmd+b`, but that's a separate PR. In this one, I'm trying to make the surrounding asterisk and back tick characters behave similar to how surrounding quotation marks and brace characters work in other code files that I write.

Reply:
Not sure whether they can really be compared as asterisk is no real quotation only markup. We also need to keep in mind that bold and italic style may be combined.

- *italic* = `*italic*` = `_italic_` 
- **bold** = `**bold**`
- ***bold italic*** = `***bold italic***` == `**_bold italic_**`

All those combinations should proberly be added and removed when pressing corresponding key binding.

Reply:
Underscore is now added.

The combinations work as you'd expect.

e.g. typing `**_` will result in `**_|_**` (where `|` is your cursor). Pressing backspace three times will remove all 6 characters.